site/public/js/polls.js

```diff
@@ -106,6 +106,81 @@ function updateDropdownStates(curr_state, cookie_key) {
     Cookies.set(cookie_key, !curr_state, { expires: expiration_date, path: '/' });
 }
 
+function addCustomResponse(pollid, base_url) {
+    const custom_response_text = document.querySelector('.custom_poll_response').value;
+    const url = `${base_url}/addCustomResponse`;
+    const fd = new FormData();
+    fd.append('csrf_token', csrfToken);
+    fd.append('poll_id', pollid);
+    fd.append('custom_response', custom_response_text);
+    $.ajax({
+        url: url,
+        type: 'POST',
+        data: fd,
+        processData: false,
+        cache: false,
+        contentType: false,
+        error: function(err) {
+            console.error(err);
+            window.alert('Something went wrong. Please try again.');
+        },
+        success: function(data) {
+            try {
+                const msg = JSON.parse(data);
+                if (msg.status !== 'success') {
+                    displayErrorMessage(msg.message);
+                }
+                else {
+                    displaySuccessMessage(msg.data.message);
+                    setTimeout(() => {
```

Comment:
I don't like reloading after one second here. I think it would be better to either reload immediately and then display the message or to not reload at all if a reload isn't needed.